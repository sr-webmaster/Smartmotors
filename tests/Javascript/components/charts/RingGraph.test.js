import { shallowMount, createLocalVue } from '@vue/test-utils'
import Component from '~/components/charts/RingGraph'

const data = [
  { label: 'a', value: 10 },
  { label: 'b', value: 20 },
  { label: 'c', value: 30 },
  { label: 'd', value: 30 },
  { label: 'e', value: 30 }
]

describe('RingGraph', () => {
  let localVue

  describe('Snapshot', () => {
    test('default', () => {
      global.SVGPathElement = jest.fn()
      localVue = createLocalVue()

      const wrapper = shallowMount(Component, {
        propsData: {
          chartData: { data }
        },
        localVue: localVue
      })

      expect(wrapper.element).toMatchSnapshot()
    })
    test('with format', () => {
      global.SVGPathElement = jest.fn()
      localVue = createLocalVue()

      const wrapper = shallowMount(Component, {
        propsData: {
          chartData: { data },
          numberFormat: "'$'#,##a"
        },
        localVue: localVue
      })

      expect(wrapper.element).toMatchSnapshot()
    })
  })

  describe('Methods', () => {
    test('render', () => {
      global.SVGPathElement = jest.fn()
      localVue = createLocalVue()

      const wrapper = shallowMount(Component, {
        propsData: {
          chartData: { data }
        },
        localVue: localVue
      })

      const chart = {
        series: {
          push: jest.fn()
        }
      }

      wrapper.setData({ chart })
      wrapper.vm.render()
      expect(chart.series.push).not.toHaveBeenCalled()
    })
  })
})
