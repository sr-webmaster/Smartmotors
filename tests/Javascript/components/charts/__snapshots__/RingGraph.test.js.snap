// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RingGraph Snapshot default 1`] = `
<div
  class="chartdiv"
  style="position: relative;"
>
  <div
    class="resize-sensor"
    dir="ltr"
    style="pointer-events: none; position: absolute; left: 0px; top: 0px; right: 0px; bottom: 0px; overflow: hidden; z-index: -1; visibility: hidden; max-width: 100%;"
  >
    <div
      class="resize-sensor-expand"
      style="pointer-events: none; position: absolute; left: 0px; top: 0px; right: 0px; bottom: 0px; overflow: hidden; z-index: -1; visibility: hidden; max-width: 100%;"
    >
      <div
        style="position: absolute; left: 0px; top: 0px; transition: 0s;"
      />
    </div>
    <div
      class="resize-sensor-shrink"
      style="pointer-events: none; position: absolute; left: 0px; top: 0px; right: 0px; bottom: 0px; overflow: hidden; z-index: -1; visibility: hidden; max-width: 100%;"
    >
      <div
        style="position: absolute; left: 0px; top: 0px; transition: 0s; width: 200%; height: 200%;"
      />
    </div>
  </div>
  <div
    style="width: 100%; height: 100%; position: relative;"
  >
    <svg
      role="group"
      style="width: 100%; height: 100%; overflow: visible;"
      version="1.1"
      xmlns="http://www.w3.org/2000/svg"
      xmlns:xlink="http://www.w3.org/1999/xlink"
    >
      <desc>
        JavaScript chart by amCharts
      </desc>
      <defs>
        <clippath
          id="id-46"
        >
          <rect
            height="NaN"
            width="NaN"
          />
        </clippath>
        <filter
          height="200%"
          id="filter-id-51"
          width="200%"
          x="-50%"
          y="-50%"
        >
          <fegaussianblur
            in="SourceGraphic"
            result="blurOut"
            stdDeviation="1.5"
          />
          <feoffset
            dx="1"
            dy="1"
            result="offsetBlur"
          />
          <feflood
            flood-color="#000000"
            flood-opacity="0.5"
          />
          <fecomposite
            in2="offsetBlur"
            operator="in"
          />
          <femerge>
            <femergenode />
            <femergenode
              in="SourceGraphic"
            />
          </femerge>
        </filter>
        <filter
          height="120%"
          id="filter-id-66"
          width="120%"
          x="-10%"
          y="-10%"
        >
          <fecolormatrix
            type="saturate"
            values="0"
          />
        </filter>
        <lineargradient
          id="gradient-id-69"
          x1="1%"
          x2="99%"
          y1="59%"
          y2="41%"
        >
          <stop
            offset="0"
            stop-color="#474758"
          />
          <stop
            offset="0.75"
            stop-color="#474758"
            stop-opacity="1"
          />
          <stop
            offset="0.755"
            stop-color="#3cabff"
            stop-opacity="1"
          />
        </lineargradient>
        <clippath
          id="id-90"
        />
        <filter
          height="200%"
          id="filter-id-95"
          width="200%"
          x="-50%"
          y="-50%"
        >
          <fegaussianblur
            in="SourceGraphic"
            result="blurOut"
            stdDeviation="1.5"
          />
          <feoffset
            dx="1"
            dy="1"
            result="offsetBlur"
          />
          <feflood
            flood-color="#000000"
            flood-opacity="0.5"
          />
          <fecomposite
            in2="offsetBlur"
            operator="in"
          />
          <femerge>
            <femergenode />
            <femergenode
              in="SourceGraphic"
            />
          </femerge>
        </filter>
      </defs>
      <g>
        <g
          fill="#ffffff"
          fill-opacity="0"
        >
          <rect />
        </g>
        <g>
          <g
            clip-path="url(\\"#id-46\\")"
            role="widget"
          >
            <g>
              <g>
                <g>
                  <g>
                    <g>
                      <g>
                        <g>
                          <g
                            role="group"
                          >
                            <g>
                              <g
                                clip-path="url(\\"#id-90\\")"
                              >
                                <g />
                              </g>
                              <g />
                              <g>
                                <g />
                              </g>
                              <g>
                                <g />
                              </g>
                              <g>
                                <g />
                              </g>
                            </g>
                          </g>
                        </g>
                      </g>
                      <g>
                        <g>
                          <g>
                            <g />
                          </g>
                        </g>
                      </g>
                    </g>
                  </g>
                  <g
                    role="group"
                  >
                    <g />
                  </g>
                </g>
              </g>
            </g>
          </g>
          <g>
            <g>
              <g
                opacity="0"
                role="tooltip"
                visibility="hidden"
              >
                <g
                  fill="#ffffff"
                  fill-opacity="0.9"
                  filter="url(\\"#filter-id-51\\")"
                  stroke="#ffffff"
                  stroke-opacity="1"
                  stroke-width="1"
                  style="pointer-events: none;"
                >
                  <path />
                </g>
                <g>
                  <g
                    fill="#ffffff"
                    style="pointer-events: none;"
                  >
                    <g
                      transform="translate(0,7)"
                    />
                  </g>
                </g>
              </g>
              <g
                visibility="hidden"
              >
                <g
                  fill="#ffffff"
                  opacity="1"
                >
                  <rect />
                </g>
                <g>
                  <g>
                    <g>
                      <g
                        fill="#f3f3f3"
                        fill-opacity="0.8"
                        stroke-opacity="1"
                      >
                        <g>
                          <g />
                        </g>
                      </g>
                      <g
                        fill="#000000"
                        fill-opacity="0.2"
                        stroke-opacity="1"
                      >
                        <g>
                          <g />
                        </g>
                      </g>
                      <g
                        fill="#000000"
                        fill-opacity="0.4"
                      >
                        <g />
                      </g>
                    </g>
                  </g>
                </g>
              </g>
              <g
                aria-labelledby="id-66-title"
                filter="url(\\"#filter-id-66\\")"
                opacity="0.3"
                style="cursor: pointer;"
              >
                <g
                  fill="#ffffff"
                  opacity="0"
                >
                  <rect />
                </g>
                <g>
                  <g
                    fill="none"
                    shape-rendering="auto"
                    stroke="#3cabff"
                    stroke-opacity="1"
                    stroke-width="1.7999999999999998"
                  >
                    <path
                      d=" M15,15  C17.4001,15 22.7998,15.0001 27,15 C31.2002,14.9999 33.2999,6 36,6 C38.7001,6 38.6999,10.5 40.5,10.5 C42.3001,10.5 42.2999,6 45,6 C47.7001,6 50.9999,14.9999 54,15 C57.0002,15.0001 58.7999,15 60,15"
                    />
                  </g>
                  <g
                    fill="none"
                    shape-rendering="auto"
                    stroke="url(\\"#gradient-id-69\\")"
                    stroke-opacity="1"
                    stroke-width="1.7999999999999998"
                  >
                    <path
                      d=" M6,15  C8.2501,15 9.7498,15.0001 15,15 C20.2502,14.9999 20.7748,3.6 27,3.6 C33.2252,3.6 33.8998,14.9999 39.9,15 C45.9002,15.0001 45.9748,15 51,15 C56.0252,15 57.7499,15 60,15"
                    />
                  </g>
                </g>
                <title
                  id="id-66-title"
                >
                  Chart created using amCharts library
                </title>
              </g>
              <g
                opacity="0"
                role="tooltip"
                visibility="hidden"
              >
                <g
                  fill="#ffffff"
                  fill-opacity="0.9"
                  filter="url(\\"#filter-id-95\\")"
                  stroke="#ffffff"
                  stroke-opacity="1"
                  stroke-width="1"
                  style="pointer-events: none;"
                >
                  <path />
                </g>
                <g>
                  <g
                    fill="#ffffff"
                    style="pointer-events: none;"
                  >
                    <g
                      transform="translate(0,7)"
                    />
                  </g>
                </g>
              </g>
            </g>
          </g>
        </g>
      </g>
    </svg>
  </div>
</div>
`;

exports[`RingGraph Snapshot with format 1`] = `
<div
  class="chartdiv"
  style="position: relative;"
>
  <div
    class="resize-sensor"
    dir="ltr"
    style="pointer-events: none; position: absolute; left: 0px; top: 0px; right: 0px; bottom: 0px; overflow: hidden; z-index: -1; visibility: hidden; max-width: 100%;"
  >
    <div
      class="resize-sensor-expand"
      style="pointer-events: none; position: absolute; left: 0px; top: 0px; right: 0px; bottom: 0px; overflow: hidden; z-index: -1; visibility: hidden; max-width: 100%;"
    >
      <div
        style="position: absolute; left: 0px; top: 0px; transition: 0s;"
      />
    </div>
    <div
      class="resize-sensor-shrink"
      style="pointer-events: none; position: absolute; left: 0px; top: 0px; right: 0px; bottom: 0px; overflow: hidden; z-index: -1; visibility: hidden; max-width: 100%;"
    >
      <div
        style="position: absolute; left: 0px; top: 0px; transition: 0s; width: 200%; height: 200%;"
      />
    </div>
  </div>
  <div
    style="width: 100%; height: 100%; position: relative;"
  >
    <svg
      role="group"
      style="width: 100%; height: 100%; overflow: visible;"
      version="1.1"
      xmlns="http://www.w3.org/2000/svg"
      xmlns:xlink="http://www.w3.org/1999/xlink"
    >
      <desc>
        JavaScript chart by amCharts
      </desc>
      <defs>
        <clippath
          id="id-127"
        >
          <rect
            height="NaN"
            width="NaN"
          />
        </clippath>
        <filter
          height="200%"
          id="filter-id-132"
          width="200%"
          x="-50%"
          y="-50%"
        >
          <fegaussianblur
            in="SourceGraphic"
            result="blurOut"
            stdDeviation="1.5"
          />
          <feoffset
            dx="1"
            dy="1"
            result="offsetBlur"
          />
          <feflood
            flood-color="#000000"
            flood-opacity="0.5"
          />
          <fecomposite
            in2="offsetBlur"
            operator="in"
          />
          <femerge>
            <femergenode />
            <femergenode
              in="SourceGraphic"
            />
          </femerge>
        </filter>
        <filter
          height="120%"
          id="filter-id-147"
          width="120%"
          x="-10%"
          y="-10%"
        >
          <fecolormatrix
            type="saturate"
            values="0"
          />
        </filter>
        <lineargradient
          id="gradient-id-150"
          x1="1%"
          x2="99%"
          y1="59%"
          y2="41%"
        >
          <stop
            offset="0"
            stop-color="#474758"
          />
          <stop
            offset="0.75"
            stop-color="#474758"
            stop-opacity="1"
          />
          <stop
            offset="0.755"
            stop-color="#3cabff"
            stop-opacity="1"
          />
        </lineargradient>
        <clippath
          id="id-171"
        />
        <filter
          height="200%"
          id="filter-id-176"
          width="200%"
          x="-50%"
          y="-50%"
        >
          <fegaussianblur
            in="SourceGraphic"
            result="blurOut"
            stdDeviation="1.5"
          />
          <feoffset
            dx="1"
            dy="1"
            result="offsetBlur"
          />
          <feflood
            flood-color="#000000"
            flood-opacity="0.5"
          />
          <fecomposite
            in2="offsetBlur"
            operator="in"
          />
          <femerge>
            <femergenode />
            <femergenode
              in="SourceGraphic"
            />
          </femerge>
        </filter>
      </defs>
      <g>
        <g
          fill="#ffffff"
          fill-opacity="0"
        >
          <rect />
        </g>
        <g>
          <g
            clip-path="url(\\"#id-127\\")"
            role="widget"
          >
            <g>
              <g>
                <g>
                  <g>
                    <g>
                      <g>
                        <g>
                          <g
                            role="group"
                          >
                            <g>
                              <g
                                clip-path="url(\\"#id-171\\")"
                              >
                                <g />
                              </g>
                              <g />
                              <g>
                                <g />
                              </g>
                              <g>
                                <g />
                              </g>
                              <g>
                                <g />
                              </g>
                            </g>
                          </g>
                        </g>
                      </g>
                      <g>
                        <g>
                          <g>
                            <g />
                          </g>
                        </g>
                      </g>
                    </g>
                  </g>
                  <g
                    role="group"
                  >
                    <g />
                  </g>
                </g>
              </g>
            </g>
          </g>
          <g>
            <g>
              <g
                opacity="0"
                role="tooltip"
                visibility="hidden"
              >
                <g
                  fill="#ffffff"
                  fill-opacity="0.9"
                  filter="url(\\"#filter-id-132\\")"
                  stroke="#ffffff"
                  stroke-opacity="1"
                  stroke-width="1"
                  style="pointer-events: none;"
                >
                  <path />
                </g>
                <g>
                  <g
                    fill="#ffffff"
                    style="pointer-events: none;"
                  >
                    <g
                      transform="translate(0,7)"
                    />
                  </g>
                </g>
              </g>
              <g
                visibility="hidden"
              >
                <g
                  fill="#ffffff"
                  opacity="1"
                >
                  <rect />
                </g>
                <g>
                  <g>
                    <g>
                      <g
                        fill="#f3f3f3"
                        fill-opacity="0.8"
                        stroke-opacity="1"
                      >
                        <g>
                          <g />
                        </g>
                      </g>
                      <g
                        fill="#000000"
                        fill-opacity="0.2"
                        stroke-opacity="1"
                      >
                        <g>
                          <g />
                        </g>
                      </g>
                      <g
                        fill="#000000"
                        fill-opacity="0.4"
                      >
                        <g />
                      </g>
                    </g>
                  </g>
                </g>
              </g>
              <g
                aria-labelledby="id-147-title"
                filter="url(\\"#filter-id-147\\")"
                opacity="0.3"
                style="cursor: pointer;"
              >
                <g
                  fill="#ffffff"
                  opacity="0"
                >
                  <rect />
                </g>
                <g>
                  <g
                    fill="none"
                    shape-rendering="auto"
                    stroke="#3cabff"
                    stroke-opacity="1"
                    stroke-width="1.7999999999999998"
                  >
                    <path
                      d=" M15,15  C17.4001,15 22.7998,15.0001 27,15 C31.2002,14.9999 33.2999,6 36,6 C38.7001,6 38.6999,10.5 40.5,10.5 C42.3001,10.5 42.2999,6 45,6 C47.7001,6 50.9999,14.9999 54,15 C57.0002,15.0001 58.7999,15 60,15"
                    />
                  </g>
                  <g
                    fill="none"
                    shape-rendering="auto"
                    stroke="url(\\"#gradient-id-150\\")"
                    stroke-opacity="1"
                    stroke-width="1.7999999999999998"
                  >
                    <path
                      d=" M6,15  C8.2501,15 9.7498,15.0001 15,15 C20.2502,14.9999 20.7748,3.6 27,3.6 C33.2252,3.6 33.8998,14.9999 39.9,15 C45.9002,15.0001 45.9748,15 51,15 C56.0252,15 57.7499,15 60,15"
                    />
                  </g>
                </g>
                <title
                  id="id-147-title"
                >
                  Chart created using amCharts library
                </title>
              </g>
              <g
                opacity="0"
                role="tooltip"
                visibility="hidden"
              >
                <g
                  fill="#ffffff"
                  fill-opacity="0.9"
                  filter="url(\\"#filter-id-176\\")"
                  stroke="#ffffff"
                  stroke-opacity="1"
                  stroke-width="1"
                  style="pointer-events: none;"
                >
                  <path />
                </g>
                <g>
                  <g
                    fill="#ffffff"
                    style="pointer-events: none;"
                  >
                    <g
                      transform="translate(0,7)"
                    />
                  </g>
                </g>
              </g>
            </g>
          </g>
        </g>
      </g>
    </svg>
  </div>
</div>
`;
